src/frontend/apps/lti_site/utils/conversejs/converse-plugins/marshaJoinDiscussionPlugin/index.ts

```diff
@@ -104,13 +106,25 @@ const addMarshaJoinDiscussionPlugin = (xmpp: XMPP, video: Video) =>
           return;
         }
 
+        let jitsiInfo: VideoJitsiConnectionInfos;
+        if(aVideo.live_info.jitsi?.hasOwnProperty('token')) {
+          // When the original jitsi info has a token property, it means
+          // jitsi needs a token to connect to it. This token must be refeshed
+          // without moderation permission and a valid lifetime.
+          jitsiInfo = await fetchJitsiInfo(aVideo)
+        } else {
+          // Otherwise, jitsi is in anonymous mode, no more info to fetch on the api,
+          // all already existing info can be send to the student.
+          jitsiInfo = aVideo.live_info.jitsi!;
+        }
```

Comment:
```suggestion
        if(aVideo.live_info.jitsi && aVideo.live_info.jitsi.hasOwnProperty('token')) {
          // When the original jitsi info has a token property, it means
          // jitsi needs a token to connect to it. This token must be refeshed
          // without moderation permission and a valid lifetime.
          jitsiInfo = await fetchJitsiInfo(aVideo)
        } else {
          // Otherwise, jitsi is in anonymous mode, no more info to fetch on the api,
          // all already existing info can be send to the student.
          jitsiInfo = aVideo.live_info.jitsi;
        }
```